src/multi_vector_simulator/C1_verification.py

```diff
@@ -766,3 +769,52 @@ def check_for_sufficient_assets_on_busses(dict_values):
                 f"so that the bus is not a dead end should be two, excluding the excess sink. "
                 f"These are the connected assets: {asset_string}"
             )
+
+
+def check_energy_system_can_fulfill_max_demand(dict_values):
+    r"""
+    Logs a logging.warning message if the aggregated installed capacity (of non-optimizable
+    assets) and maximum capacity (of optimizable assets) of all conversion, generation and storage
+    assets is smaller than the maximum demand, for each bus. Check passes if the maximum capacity
+    of an asset is set to None for optimization.
+
+    Parameters
+    ----------
+    dict_values : dict
+        Contains all input data of the simulation.
+
+    Returns
+    -------
+    Indirectly, logs a logging.warning message if the installed and maximum capacities of
+    conversion/generation/storage assets are less than the maximum demand, for each bus.
+    """
+    for bus in dict_values[ENERGY_BUSSES]:
+        #todo: make sure while loop is executing properly
+        #todo: find a way to combine lines 802-815
+        skip_test = False
+        while skip_test is False:
+            for item in dict_values[ENERGY_BUSSES][bus][ASSET_DICT]:
+                peak_demand = 0
+                peak_generation = 0
+                if item in dict_values[ENERGY_CONSUMPTION] and \
+                        dict_values[ENERGY_CONSUMPTION][item][DISPATCHABILITY] is False:
+                    peak_demand += max(dict_values[ENERGY_CONSUMPTION][item][TIMESERIES])
+                if item in dict_values[ENERGY_CONVERSION]:
+                    peak_generation += dict_values[ENERGY_CONVERSION][INSTALLED_CAP]
+                    if dict_values[ENERGY_CONVERSION][MAXIMUM_CAP] is None:
+                        skip_test = True
+                    else:
+                        peak_generation += dict_values[ENERGY_CONVERSION][MAXIMUM_CAP]
+                if item in dict_values[ENERGY_PRODUCTION]:
+                    peak_generation += dict_values[ENERGY_PRODUCTION][INSTALLED_CAP]
+                    if dict_values[ENERGY_PRODUCTION][MAXIMUM_CAP] is None:
+                        skip_test = True
+                    else:
+                        peak_generation += dict_values[ENERGY_PRODUCTION][MAXIMUM_CAP]
```

Comment:
With avoiding the `while` loop, I would suggest this instead, as it is easier to test:
```suggestion
                for group in [ENERGY_CONVERSION, ENERGY_PRODUCTION]
                    if item in dict_values[group]:
                        peak_generation += dict_values[group][item][INSTALLED_CAP][VALUE]
                        if dict_values[group][item][MAXIMUM_CAP][VALUE] is None:
                             skip_test = True
                        else:
                             peak_generation += dict_values[group][item][MAXIMUM_CAP][VALUE]
```
I also changed the way the dict is read (group -> item (asset) -> parameter -> value)

Reply:
Great! This was what I was looking into doing but hadn't properly implemented it yet